packages/syft/src/syft/types/errors.py

```diff
@@ -0,0 +1,51 @@
+# stdlib
+from typing import Literal
+
+# relative
+from ..service.user.user_roles import ServiceRole
+
+SyftErrorCodes = Literal[
+    "not-found",
+    "not-permitted",
+    "database-error",
+    "serde-serialization-error",
+    "serde-deserialization-error",
+    "stash-error",
+    "worker-invalid-pool",
+    "usercode-not-approved",
+    "usercode-bad-input-policy",
+    "usercode-bad-output-policy",
+]
+
+
```

Comment:
I think having our own exception types is a really good change over the current Result type, it moves our error handling a lot to the standard python way of doing it.

I think this is something we should zoom in on as main part of the error handling re-vamp: this is how you're supposed to specify and propagate errors in Python, the result is just some nice on top to reduce boilerplate big try-catch blocks